website/src/components/comment-section.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { ReactNode, SetStateAction } from "react"
+import { useState } from "react"
+import * as Dailp from "src/graphql/dailp"
+import { TranslatedParagraph } from "src/segment"
+import * as css from "./comment-section.css"
+
+export const CommentSection = (p: {
+  parent: Dailp.FormFieldsFragment | TranslatedParagraph
+}) => {
+  return (
+    <div>
+      {p.parent.__typename === "DocumentParagraph" ? (
+        <ParagraphCommentSection paragraph={p.parent} />
+      ) : p.parent.__typename === "AnnotatedForm" ? (
+        <WordCommentSection word={p.parent} />
+      ) : (
+        ""
```

Comment:
`parent` cannot be `undefined`, `null`, or some other type. You can delete this embedded ternary to leave you with the following:

```
{p.parent.__typename === "DocumentParagraph" ? (
         <ParagraphCommentSection paragraph={p.parent} />
       ) : <WordCommentSection word={p.parent} />)}

Reply:
when I try to delete the embedded ternary I get this error: ```
Type 'FormFieldsFragment | ({ readonly __typename?: "DocumentParagraph" | undefined; } & Pick<DocumentParagraph, "id" | "index" | "translation"> & { ...; })' is not assignable to type 'FormFieldsFragment'.
  Type '{ readonly __typename?: "DocumentParagraph" | undefined; } & Pick<DocumentParagraph, "id" | "index" | "translation"> & { readonly source: readonly (({ readonly __typename: "AnnotatedForm"; } & Pick<...> & { ...; }) | { ...; })[]; readonly comments: readonly ({ ...; } & Pick<...>)[]; }' is not assignable to type 'FormFieldsFragment'.
    Type '{ readonly __typename?: "DocumentParagraph" | undefined; } & Pick<DocumentParagraph, "id" | "index" | "translation"> & { readonly source: readonly (({ readonly __typename: "AnnotatedForm"; } & Pick<...> & { ...; }) | { ...; })[]; readonly comments: readonly ({ ...; } & Pick<...>)[]; }' is not assignable to type '{ readonly __typename?: "AnnotatedForm" | undefined; }'.
      Types of property '__typename' are incompatible.
        Type '"DocumentParagraph" | undefined' is not assignable to type '"AnnotatedForm" | undefined'.
```

Reply:
It looks like typescript is not being nice to you. Sometimes nested fields will be like this when you try to do a type guard (checking if a value is of a certain type, like you do when you check the value of `__typename`) on a literal. You might see if this goes away if instead of taking in an object of params like this:
```
(p: {...prop type}): ReactElement
```
You instead "destructure" your props like this:
```
({parent}: {... prop type}): ReactElement
```

Here are some docs on "destructuring" in JS if that is new for you: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

Reply:
We may also fix this if we upgrade typescript somehow. This should work.